src/cls/rbd/cls_rbd.cc

```diff
@@ -5767,346 +5795,1478 @@ int image_snapshot_set_copy_progress(cls_method_context_t hctx,
   return 0;
 }
 
-} // namespace mirror
+int group_list(cls_method_context_t hctx,
```

Comment:
Nit : I think it would be better to add the new functions after the existing mirror functions.